### ISSUE TITLE
Use Classes for *All* Entities?

### ISSUE THREAD

**Issue by ebollens** [ucla-oarc-web/WebBlocks#98]
Should we require classes for all our entities so we don't subsume someone's intended style when they lay WebBlocks in? The cases I can think of right off the bat are:

``` html
<table>
    <!-- ... -->
</table>

<ul>
    <!-- ... -->
</ul>

<ol>
    <!-- ... -->
</ol>
```

If we used classes, they would be:

``` html
<table class="table">
    <!-- ... -->
</table>

<ul class="list">
    <!-- ... -->
</ul>

<ol class="list">
    <!-- ... -->
</ol>
```

@atsengucla @ccheung @loganfranken thoughts?

If we think we should always use class names, then I can refactor the existing elements. This pertains to #72 as well. I think I'm leaning towards always using class names. That ensures no unintended consequences by laying something in (except what the adapter does on its own - as Bootstrap uses a class for table but not one for lists).

**Comment by ebollens** [comment]
If people wanted to have this "subsume" (such as styling `table` and `ul` without class names), we could provide an extension that does this. I know I'd use that extension if I was starting fresh, but I probably wouldn't use that extension if I was working on an existing site.

**Comment by alicetseng** [comment]
Using explicit classes is the safest move. I'm for it. We should strive to do the least amount of un-necessary, unintended over-writes. It's the pattern that we see in the shift from "browser reset code" to normalize.

**Comment by ebollens** [comment]
I'm in concurrence. @loganfranken thoughts?

**Comment by loganfranken** [comment]
Yup, I'm for it too!

**Comment by ebollens** [comment]
Added a **development** tag to this issue and I will make these changes in the existing implementations and prototypes.

**Comment by ebollens** [comment]
For minimizing the amount of CSS required, I've used the `.table` selector rather than the `table.table` selector. This is to avoid the extra five bytes that come with specifying the element itself. However, by our prototype, we've defined the semantics, and unless we use `.table` is some context besides with the `table` element, this choice should be just fine.

Also, as a note, check out that commit e875852. Shows one of the advantages to using definitions that invoke mixins. There was no need to make any changes to any of the adapters :-)
